src/common/connection.ts

```diff
@@ -36,29 +36,67 @@ export interface ConnectionOptions {
  */
 export type ConnectionUserOptions = Partial<ConnectionOptions>
 
+/**
+ * Ledger is used to store and reference ledger information that has been
+ * captured by the Connection class.
+ */
+class Ledger {
+  private availableVersions = new RangeSet()
+  latestVersion: null | number = null
+  feeBase: null | number = null
+  feeRef: null | number = null
+
+  hasVersions(lowVersion: number, highVersion: number): boolean {
+    return this.availableVersions.containsRange(lowVersion, highVersion)
+  }
+
+  hasVersion(version: number): boolean {
+    return this.availableVersions.containsValue(version)
+  }
+
+  update(data: {
+    ledger_index?: string
+    validated_ledgers?: string
```

Comment:
Nit: this class is mixing camel case and snake case.

Reply:
I think we can use snake case here if it is coming directly from rippled: https://xrpl.org/subscribe.html#ledger-stream

Reply:
yea, the idea is that this just takes any Rippled response object, and checks it for these properties. will clarify this in docs or argument name

Reply:
Done, lmk if that clarifies whats going on